src/antsibull_docs/schemas/app_context.py

```diff
@@ -8,11 +8,10 @@
 import pydantic as p
 from antsibull_core.schemas.context import AppContext as CoreAppContext
 from antsibull_core.schemas.validators import convert_bool
-
-from antsibull_docs._pydantic_compat import Field
+from pydantic import Field
 
 #: Valid choices for the docs parsing backend
-DOC_PARSING_BACKEND_CHOICES_F = Field("auto", regex="^(auto|ansible-core-2\\.13)$")
+DOC_PARSING_BACKEND_CHOICES_F = Field("auto", pattern="^(auto|ansible-core-2\\.13)$")
```

Comment:
While the Rust regex create used by pydantic v2 to validate regexes is more efficient than Python's re that was used before, we should switch these to use `Literal` instead, either here in another PR.

Reply:
I agree. But let's do that in another PR.

Reply:
Hmm, I tried replacing `doc_parsing_backend: str = DOC_PARSING_BACKEND_CHOICES_F` with `doc_parsing_backend: str = t.Literal["auto", "ansible-core-2.13"]`, and fed it a config file with an invalid value. With `t.Literal` it did **not** complain about the invalid value.

Reply:
Hmm, and I tried it again with the CollectionMetadata schema from antsibull-core, and *there* it works... I have to try again :)

Reply:
Ok, I did a stupid mistake. It works fine...